python/pfs/drp/stella/reduceArcRefSpecTask.py

```diff
@@ -206,4 +207,27 @@ def run(self, expRefList, butler, refSpec=None, lineList=None, immediate=True):
                 plt.close(fig)
                 fig.clf()
 
+            #
+            # Do the I/O using a trampoline object PfsArmIO (to avoid adding butler-related details
+            # to the datamodel product)
+            #
+            # This is a bit messy as we need to include the pfsConfig file in the pfsArm file
+            #
+            dataId = arcRef.dataId
+
+            md = arcExp.getMetadata().toDict()
+            key = "PFSCONFIGID"
+            if key in md:
+                pfsConfigId = md[key]
+            else:
+                self.log.info('No pfsConfigId is present in postISRCCD file for dataId %s' %
+                              str(dataId.items()))
+                pfsConfigId = 0x0
+                                                                                                 
+            pfsConfig = butler.get("pfsConfig", pfsConfigId=pfsConfigId, dateObs=dataId["dateObs"])
+
+            pfsArm = spectrumSetToPfsArm(pfsConfig, spectrumSetFromProfile,
+                                         dataId["visit"], dataId["spectrograph"], dataId["arm"])
+            butler.put(PfsArmIO(pfsArm), 'pfsArm', dataId)
+
```

Comment:
This looks like code copied from reduceArcTask.py; if so, why didn't you factor it out into a function?  It looks like the standard way of saving a pfsArm file and will be used for all data reduction tasks (not just arcs).